sdk/log/simple.go

```diff
@@ -55,7 +98,20 @@ func (s *SimpleProcessor) OnEmit(ctx context.Context, r *Record) error {
 		simpleProcRecordsPool.Put(records)
 	}()
 
-	return s.exporter.Export(ctx, *records)
+	err := s.exporter.Export(ctx, *records)
+
+	if s.selfObservabilityEnabled {
```

Comment:
I think this part of the code would be more appropriate if placed before the following code. Additionally, using the defer approach for reporting would result in higher code readability.

```go
if s.exporter == nil {
```